addons/mail/tests/discuss/test_discuss_channel.py

```diff
@@ -594,6 +594,25 @@ def test_channel_write_should_send_notification(self):
         ):
             channel.name = "test test"
 
+    def test_channel_mention_should_send_message_in_channel(self):
+        channel = self.env['discuss.channel'].create({"name": "Test Channel", "channel_type": "channel"})
+        partner = self.env["res.partner"].create({
+            "name": "Test Partner",
+            "email": "testpartner@odoo.com",
+        })
+        # Message Post
+        message = partner.message_post(body="Hello", channel_ids=[channel.id], message_type="comment", subtype_xmlid="mail.mt_comment")
+        # Check that the mentioned channel containt a message with id of channel
```

Comment:
Realistically, body should contain the channel mention. "Hello" is not a message that should make a channel mention.

This works because you rely on `channel_id` param, which is there to help python processing it. That's because the linkifying of mention is made client-side, and this param prevents python code to parse body to find these channel mentions.

I don't mind this test in python that leaks this implementation detail, but at least make body realistic. This is the most important param that determines whether the message contains a mention or not.